xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -469,33 +472,36 @@ void CGUIDialogVideoInfo::OnSearch(std::string& strSearch)
 
   if (items.Size())
   {
-    CGUIDialogSelect* pDlgSelect = static_cast<CGUIDialogSelect*>(g_windowManager.GetWindow(WINDOW_DIALOG_SELECT));
-    pDlgSelect->Reset();
-    pDlgSelect->SetHeading(CVariant{283});
-
-    CVideoThumbLoader loader;
-    for (int i = 0; i < items.Size(); i++)
+    CGUIDialogSelect* pDlgSelect = dynamic_cast<CGUIDialogSelect*>(g_windowManager.GetWindow(WINDOW_DIALOG_SELECT));
+    if(pDlgSelect)
     {
-      if (items[i]->HasVideoInfoTag() &&
-          items[i]->GetVideoInfoTag()->m_playCount > 0)
-        items[i]->SetLabel2(g_localizeStrings.Get(16102));
+      pDlgSelect->Reset();
+      pDlgSelect->SetHeading(CVariant{283});
 
-      loader.LoadItem(items[i].get());
-      pDlgSelect->Add(*items[i]);
-    }
+      CVideoThumbLoader loader;
+      for (int i = 0; i < items.Size(); i++)
+      {
+        if (items[i]->HasVideoInfoTag() &&
+            items[i]->GetVideoInfoTag()->m_playCount > 0)
+          items[i]->SetLabel2(g_localizeStrings.Get(16102));
 
-    pDlgSelect->SetUseDetails(true);
-    pDlgSelect->Open();
+        loader.LoadItem(items[i].get());
+        pDlgSelect->Add(*items[i]);
+      }
 
-    int iItem = pDlgSelect->GetSelectedItem();
-    if (iItem < 0)
-      return;
+      pDlgSelect->SetUseDetails(true);
+      pDlgSelect->Open();
 
-    CFileItem* pSelItem = new CFileItem(*items[iItem]);
+      int iItem = pDlgSelect->GetSelectedItem();
+      if (iItem < 0)
+        return;
 
-    OnSearchItemFound(pSelItem);
+      CFileItem* pSelItem = new CFileItem(*items[iItem]);
 
-    delete pSelItem;
+      OnSearchItemFound(pSelItem);
+
+      delete pSelItem;
```

Comment:
Why not using a smart pointer here (CFileItemPtr). Saves you the manual delete. maybe this can be improved at other places in this file (not checked)?